newdle/client/src/components/AnswerCalendar.js

```diff
@@ -0,0 +1,175 @@
+import _ from 'lodash';
+import {Grid} from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import React from 'react';
+import moment from 'moment';
+import {serializeDate, toMoment} from '../util/date';
+import {useSelector} from 'react-redux';
+import AnswerCalendarDay from './AnswerCalendarDay';
+import {getNewdleTimeslots, getNewdleDuration, getAnswers} from '../selectors';
+import {addAnswer, removeAnswer} from '../actions';
+import styles from './Answer.module.scss';
+
+const OVERFLOW_HEIGHT = 0.5;
+const MIN_HEIGHT = 5;
+
+function calculateHeight(start, end, minHour, maxHour) {
+  let startMins = start.hours() * 60 + start.minutes();
+  let endMins = end.hours() * 60 + end.minutes();
+
+  if (startMins < minHour * 60) {
+    startMins = minHour * 60;
+  }
+  if (endMins > maxHour * 60) {
+    endMins = maxHour * 60;
+  }
+  const height = ((endMins - startMins) / ((maxHour - minHour) * 60)) * 100;
+  return height > MIN_HEIGHT ? height : MIN_HEIGHT;
+}
+
+function calculatePosition(start, minHour, maxHour) {
+  const spanMins = (maxHour - minHour) * 60;
+  let startMins = start.hours() * 60 + start.minutes() - minHour * 60;
+
+  if (startMins < 0) {
+    startMins = 0;
+  }
+
+  const position = (startMins / spanMins) * 100;
+  return position < 100 ? position : 100 - OVERFLOW_HEIGHT;
+}
+
+function groupOverlaps(options) {
+  // sort options
+  const sortedOptions = _.orderBy(options, ['pos'], ['asc']);
+  // mark items that overlap with their successors
+  let clusterId = 0;
+  const clusteredOptions = sortedOptions.map((option, index, sortedOptions) => {
+    if (
+      sortedOptions[index - 1] &&
+      toMoment(option.startTime, 'HH:mm').isSameOrAfter(
+        toMoment(sortedOptions[index - 1].endTime, 'HH:mm')
+      )
+    ) {
+      clusterId += 1;
+    }
+    return {...option, clusterId};
+  });
+  // group overlapping options
+  return Object.values(_.groupBy(clusteredOptions, c => c.clusterId));
+}
+
+function getAnswerProps(slot, answer) {
+  if (answer === 'available') {
+    return {
+      icon: 'check square outline',
+      action: addAnswer(slot, 'ifneedbe'),
```

Comment:
I had to look at this twice to realize that it's not a bug... I'd rather make `action` an arrow function returning this and then `dispatch(option.action())` instead of `dispatch(option.action)`